app/client/src/store/actions.ts

```diff
@@ -45,17 +45,33 @@ export default {
   },
   async runPoppunk(context: ActionContext<RootState, RootState>) {
     const { state, commit } = context;
-    const jsonSketches = {} as { [key: string]: Record<string, never> };
-    // join all file hashes to create a project hash
-    const phash = Md5.hashStr(Object.keys(state.results.perIsolate).sort().join());
+    // generate filenameMapping that the backend can use to replace
+    // filehashes with filenames in results.
+    // Also generate a string of ordered hashes and corresponding filenames
+    // to be used to generate a projecthash that is unique to this combination
+    // of file contents and filenames
+    const filenameMapping = {} as { [key: string]: string | undefined };
```

Comment:
```suggestion
    const filenameMapping = {} as Record<string, string | undefined>;
```
A little more readable maybe? Or we often define a `Dict` type as it's generally useful: `export type Dict<V> = { [k: string]: V }`  Could consider doing likewise?
Why would a value in the mapping ever be undefined?

Reply:
when removing undefined, line 56 would give an typescript error for `filenameMapping[filehash]`: `Type 'string | undefined' is not assignable to type 'string'`. I guess it is undefined for the empty filenameMapping without any elements in it?

Reply:
Yes, this should always have a value, since the action can only be triggered when files have been uploaded. However, adding '!' causes `warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion` on linting, shall I just switch that warning off?